source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRViewController.mm

```diff
@@ -58,7 +58,43 @@ - (void)render
     UIView *view = self.view;
     view.frame = _guideFrame;
     NSMutableArray *inputs = [[NSMutableArray alloc] init];
-
+    
+    std::string backgroundImage = _adaptiveCard->GetBackgroundImage();
+    NSString* imgUrl = nil;
+    if(backgroundImage.size())
+        imgUrl = [[NSString alloc] initWithCString:backgroundImage.c_str() encoding:NSUTF8StringEncoding];
+    if (imgUrl)
+    {
+        NSURL *url = [NSURL URLWithString:imgUrl];
+        UIImage *img = [UIImage imageWithData:[NSData dataWithContentsOfURL:url]];
+        UIImageView *imgView = [[UIImageView alloc] initWithImage:img];
+        [view addSubview:imgView];
+        [view sendSubviewToBack:imgView];
+        [NSLayoutConstraint activateConstraints:
+         @[[imgView.trailingAnchor constraintEqualToAnchor:view.trailingAnchor],
+           [imgView.topAnchor constraintEqualToAnchor:view.topAnchor],
+           [imgView.trailingAnchor constraintEqualToAnchor:view.trailingAnchor],
+           ]];
+    }
+    
+    ContainerStyle style = (_hostConfig->adaptiveCard.allowCustomStyle)? _adaptiveCard->GetStyle() : _hostConfig->actions.showCard.style;
+    if(style != ContainerStyle::None)
+    {
+        long num = 0;
+        if(style == ContainerStyle::Emphasis)
+        {
+            num = std::stoul(_hostConfig->containerStyles.emphasisPalette.backgroundColor.substr(1), nullptr, 16);
```

Comment:
>num = std::stoul(_hostConfig->containerStyles.emphasisPalette.backgroundColor.substr(1), nullptr, 16) [](start = 12, length = 101)

Is there any reason not to transform each substring directly?

ie:

std::string colorString = _hostConfig->containerStyles.emphasisPalette.backgroudnColor;

long alpha = std::stol(colorString.substr(1,2));
long red = std::stol(colorString.substr(3,2));
etc...

That way you avoid the bit operations that seem out of place.

Reply:
it's more efficient. it's making one function call. and bit operations are just singe instructions.

---
In reply to: [155121743](https://github.com/Microsoft/AdaptiveCards/pull/1029#discussion_r155121743) [](ancestors = 155121743)

Reply:
This is a very short string. The performance gains are minimal, while the code would become easier to read and change in the future.

---
In reply to: [155122152](https://github.com/Microsoft/AdaptiveCards/pull/1029#discussion_r155122152) [](ancestors = 155122152,155121743)

Reply:
If you feel strongly about the bit operations, you can keep the code as is, but please weigh those considerations in the future.

---
In reply to: [155122288](https://github.com/Microsoft/AdaptiveCards/pull/1029#discussion_r155122288) [](ancestors = 155122288,155122152,155121743)

Reply:
sure, i will do that. 

---
In reply to: [155122484](https://github.com/Microsoft/AdaptiveCards/pull/1029#discussion_r155122484) [](ancestors = 155122484,155122288,155122152,155121743)